modules/library/main/src/main/java/org/geotools/geometry/jts/WKTWriter2.java

```diff
@@ -0,0 +1,835 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *   (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *   (C) 2001, Vivid Solutions
+ *   
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *    
+ *    This is a port of the JTS WKTReader to handle SQL MM types such as Curve.
+ *    We have subclassed so that our implementation can be used anywhere 
+ *    a WKTReader is needed. We would of tried for more code reuse  except
+ *    the base class has reduced everything to private methods.
+ */
```

Comment:
Let's leave it alone, when I last did an IP review they had me double back and restore some original headers.